src/topology/subset_properties.lean

```diff
@@ -569,6 +569,121 @@ theorem is_irreducible.image [topological_space β] {s : set α} (H : is_irreduc
   (f : α → β) (hf : continuous_on f s) : is_irreducible (f '' s) :=
 ⟨nonempty_image_iff.mpr H.nonempty, H.is_preirreducible.image f hf⟩
 
+lemma subtype.preirreducible_space {s : set α} (h : is_preirreducible s) :
+  preirreducible_space s :=
+{ is_preirreducible_univ :=
+  begin
+    intros u v hu hv hsu hsv,
+    rw is_open_induced_iff at hu hv,
+    rcases hu with ⟨u, hu, rfl⟩,
+    rcases hv with ⟨v, hv, rfl⟩,
+    rcases hsu with ⟨⟨x, hxs⟩, hxs', hxu⟩,
+    rcases hsv with ⟨⟨y, hys⟩, hys', hyv⟩,
+    rcases h u v hu hv ⟨x, hxs, hxu⟩ ⟨y, hys, hyv⟩ with ⟨z, hzs, ⟨hzu, hzv⟩⟩,
+    exact ⟨⟨z, hzs⟩, ⟨set.mem_univ _, ⟨hzu, hzv⟩⟩⟩
+  end }
+
+lemma subtype.irreducible_space {s : set α} (h : is_irreducible s) :
+  irreducible_space s :=
+{ is_preirreducible_univ :=
+  (subtype.preirreducible_space h.is_preirreducible).is_preirreducible_univ,
+  to_nonempty := h.nonempty.to_subtype }
+
+lemma is_irreducible_iff_sInter {s : set α} :
```

Comment:
Could you add docstrings to the nontrivial statements, to help the user?

Reply:
Done